security/crypto/bls12/bls12.go

```diff
@@ -350,75 +343,85 @@ func (bls *bls12Base) Combine(signatures ...hotstuff.QuorumSignature) (combined
 			}
 			g2.Add(&agg, &agg, &sig2.sig)
 		} else {
```

Comment:
If you flip the `ok` check on Line 332 to `!ok`, you could move this error return early and avoid the else part so that the main logic can be unindented for a more readable flow.